seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/config/JdbcOptions.java

```diff
@@ -35,6 +35,13 @@ public interface JdbcOptions {
     Option<String> DRIVER =
             Options.key("driver").stringType().noDefaultValue().withDescription("driver");
 
+    Option<String> DIALECT_NAME =
+            Options.key("dialect.name")
+                    .stringType()
+                    .defaultValue("default")
+                    .withDescription(
+                            "The dialect name, when use same jdbc drive, use this parameter to choose the special implement.");
```

Comment:
So if I use this config will not work right?
```
Jdbc {
    driver = com.mysql.cj.jdbc.Driver
    url = "jdbc:mysql://mysql:9030"
    user = root
    password = ""
    query = "select * from `test`.`e2e_table_source`"
    partition_column = "STRING_COL"
    dialect.name = "mysql"
  }
```
It's weird.

Reply:
yes, if we support dialect.name = "mysql" then this parameter is required to mysql connector, it will impact the current user.

and i think this feature is the special requirement, use same jdbc driver, and the default dialect can not support some feature, it can use this parameter to choose the corresponding dialect implement.  

In this case, only those user need this feature need add this parameter